corgi/tasks/pnc.py

```diff
@@ -0,0 +1,62 @@
+import logging
+from json import JSONDecodeError
+
+import requests
+from celery_singleton import Singleton
+from requests.exceptions import HTTPError
+
+from config.celery import app
+from corgi.collectors.cyclonedx import CycloneDxSbom
+from corgi.core.models import ProductVariant
+
+logger = logging.getLogger(__name__)
+
+
+@app.task(base=Singleton)
+def slow_fetch_pnc_sbom(purl: str, product_data, build_data, sbom_data) -> None:
+    logger.info("Fetching PNC SBOM %s for purl %s", sbom_data["id"], purl)
+    """Fetch a PNC SBOM from sbomer"""
+
+    # Validate the supplied product information
+    try:
+        product_variant = ProductVariant.objects.get(name=product_data["productVariant"])
```

Comment:
Based on the discussion with Justin last week in chat about variants, I think our ProductVariants don't necessarily match to ET Variants. Is there any model that maps to ET Variants directly @jasinner?

Reply:
The names of ET Variants should match what we use in the name field of the ProductVariant core models. There are 2 ways we create those models. But matching brew_tags from product_definitions with those from ET, in which case we use the ET Variant name to create the ProductVariant core model. The other way is by directly reading and creating errata_info from product_definitions. In this case if product_definitions has bad data we could get something which doesn't not exist in ET, we don't do validation. We have the data in the CollectorErrataProductVariant collector model, so we could do such validation.